addons/mail/static/src/components/discuss/discuss_tests.js

```diff
@@ -5519,6 +5519,303 @@ QUnit.test('moderation: sent message rejected in moderated channel', async funct
     );
 });
 
+QUnit.test('moderation: as moderator, pending moderation message appears in moderation box and in origin thread', async function (assert) {
+    assert.expect(5);
+
+    Object.assign(this.data.initMessaging, {
+        channel_slots: {
+            channel_channel: [{
+                id: 20,
+                channel_type: "channel",
+                name: "general",
+                moderation: true,
+            }],
+        },
+        is_moderator: true,
+        moderation_counter: 1,
+        moderation_channel_ids: [20],
+    });
+    this.data['mail.message'].records = [{
+        author_id: [2, "Someone"],
+        body: "<p>test</p>",
+        channel_ids: [20],
+        id: 100,
+        model: 'mail.channel',
+        moderation_status: 'pending_moderation',
+        need_moderation: true,
+        res_id: 20,
+    }];
+
+    await this.start();
+
+    assert.ok(
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="${
+                this.env.messaging.moderation.localId
+            }"]
+        `),
+        "should display the moderation box"
+    );
+
+    await afterNextRender(() =>
+        document.querySelector(`
+            .o_DiscussSidebar_item[data-thread-local-id="mail.thread_mail.channel_20"]
+        `).click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "the pending moderation message should be in the channel"
+    );
+    assert.strictEqual(
+        document.querySelector('.o_Message_content').textContent,
+        "test",
+        "the pending moderation message should have the correct content"
+    );
```

Comment:
I prefer checking local id in data (we will change later to use id instead), because assertion the message is displayed just based on content is weak.